### ISSUE TITLE
Polish Headers Side Panel

### ISSUE THREAD

**Issue by janodvarko** [firefox-devtools/ux#116]
The Headers side panel (in the Network panel) would deserve some UI/X clean-up
(this might be done as part of the Accordion refactoring see bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1613884

There are currently three expandable sections:
1) Response Headers
2) Request Headers
3) Request headers from upload stream

We could:
* Rename the third one to `Request Payload`
* Turn the general info at the top into an expandable section called `General`
* Move the `Edit and Resend` button to a new toolbar located at the top of the panel

WDYT? @digitarald @bomsy @fvsch 

![image](https://user-images.githubusercontent.com/37785/74822336-45d8cf00-5305-11ea-91d1-88931b58aa1c.png)

**Comment by digitarald** [comment]
There is both UI work to better align the space in this panel and UX for the information hierarchy.

From user feedback "network header tab is confusing", "a lot of information in a very compressed space – should be redone", "easy selecting and copy-pasting of request/response headers (for example: right-click header name and have the option to copy the name, value or name and value)."

Constraints that I'd like to throw in:

- Consistent raw/pretty split/toggles for all data types (headers, params, payloads, cookies)
- Consistent search in all important data types
- Obvious copy/save for whole raw/pretty sections or specific entries and their parts (mostly name/value)
- Maybe: States for pending request, streamed request/response

Questions:

- Would it make sense to move away from tabs to one accordion holding more/all sections?

**Comment by fvsch** [comment]
> Would it make sense to move away from tabs to one accordion holding more/all sections?

Many sections have sub-sections, so we'd end up with like 10 accordion items which is a bit much. And for sections like Messages (websocket) it's useful to have the full sidebar's height available.

> Consistent raw/pretty split/toggles for all data types (headers, params, payloads, cookies)

I notice that the Debugger when for a standard checkbox (in the Event Listener Breakpoints accordion header). We could use that, or make a smaller toggle control (the current one is rather big) and use it in Debugger too.

Personally for Headers I'd be happier with a single "Raw Headers" toggle (next to "Filter Headers") that affects all sections.

> Maybe: States for pending request, streamed request/response

Yes please! It's hard to figure out that a request is pending just by looking at the table row, and you're left wondering if something is broken when the details are blank.

**Comment by digitarald** [comment]
> And for sections like Messages (websocket) it's useful to have the full sidebar's height available.

Agreed, beyond WS, preview, formatted payloads and timings work better with more space. Security could link to the cert viewer or [inline it](https://bugzilla.mozilla.org/show_bug.cgi?id=1608962).

To simplify the tabs, Cookies and Params could be rolled into the main panel. They often end up empty as well.

Related, Honza and I discussed why WS' `Messages` could not be shown in `Response`.

**Comment by digitarald** [comment]
A related UX bug, improving the summary view for cached resources: https://bugzilla.mozilla.org/show_bug.cgi?id=1338065

**Comment by bomsy** [comment]
Also to add to the list of suggestions from honza above
- Can we move the filter headers to the top to be consistent with the other panels?

**Comment by fvsch** [comment]
> Can we move the filter headers to the top to be consistent with the other panels?

A small toolbar with "Filter Headers" and a toggle for "Raw Headers" would be great. It'd be consistent with the Messages tab, or with what we have in the Inspector for Computed styles, for instance:

![Screen Shot 2020-02-21 at 13 49 38](https://user-images.githubusercontent.com/243601/75035990-149d0200-54b1-11ea-844b-3897d889decc.png)

**Comment by janodvarko** [comment]
Great, I really these ideas (search box within a toolbar at the top)
Created a bug for it: https://bugzilla.mozilla.org/show_bug.cgi?id=1617167

Honza

**Comment by digitarald** [comment]
> Great, I really these ideas (search box within a toolbar at the top)

I shared @fvsch's concern on how it doesn't search through the first part.

Maybe we can find a consistent toolbar that allows filter/copy/pretty-raw-toggle?

**Comment by digitarald** [comment]
I found [Paw](paw.cloud) last week and found its UI for inspection nice and clean:

<hr>

Splits data in _Info_, _Request_ (Headers, Formatted, Raw) and _Response_ (Headers, Formatted, Raw)

<img width="612" alt="image" src="https://user-images.githubusercontent.com/8599/75295662-0f58f380-57e0-11ea-8d02-34817d2786dd.png">

_Prev/Next_ buttons in toolbar are nice affordance for navigating between ordered requests (often in REST and page load):

<img width="54" alt="image" src="https://user-images.githubusercontent.com/8599/75295686-1d0e7900-57e0-11ea-9146-eb2eae44b2cb.png">

Summarized request/response meta data links to the _Request_/_Response_ tabs: 
<img width="481" alt="image" src="https://user-images.githubusercontent.com/8599/75295734-37e0ed80-57e0-11ea-9e51-bf7db6a172ca.png">

_Raw_ tab shows full HTTP (with basic highlighting for headers):

<img width="482" alt="image" src="https://user-images.githubusercontent.com/8599/75295789-64950500-57e0-11ea-8113-e34ec74a80a2.png">

Share button to export & upload to online viewer:

<img width="224" alt="image" src="https://user-images.githubusercontent.com/8599/75295832-7f677980-57e0-11ea-92af-aea1ac93ab92.png">

**Comment by janodvarko** [comment]
> I found Paw last week and found its UI for inspection nice and clean:

I can't open the Paw link (or is it suppose to be a link?)

**Comment by digitarald** [comment]
> I can't open the Paw link (or is it suppose to be a link?)

Fixed. It is on paw.cloud :).

**Comment by digitarald** [comment]
Updating priority as @bomsy will be starting on this soon.

**Comment by digitarald** [comment]
@violasong fyi, we had some discussion today for some mock ups I made: https://www.figma.com/file/cf8pMDTvi64HWzRhQWdMnO/DevTools-Network-Details-Pane?node-id=32%3A2

**Comment by violasong** [comment]
I'm working on a mockup but here are some initial thoughts:

Could we hide the Cookies/Params/etc tabs if they're empty, or could this be confusing?

Paw is a great find! Nice example of Mac native patterns.

For copying, there are a couple things we could try:
- When selecting a row, Cmd/Ctrl-C should copy both name/value
- Double-clicking a name or value should select that whole string (currently the selection stops at any hypens, so for an example like this, it's hard to grab the whole name or value) <img width="449" alt="image" src="https://user-images.githubusercontent.com/55994/77610363-bd3ee700-6edf-11ea-9437-1d66d3ccd671.png">
- Copy button that shows up on hover at the end of each row - maybe a bit much. Could be batched up with the info button which should be on the right side.
- Raw header sections could have an always visible copy button

Some little consistency issues that could be fixed:
- Accordion header text can currently be selected. Also, it has a blue background on hover.
<img width="437" alt="image" src="https://user-images.githubusercontent.com/55994/77610427-ea8b9500-6edf-11ea-9edb-620a6f364e38.png">

> States for pending request, streamed request/response

Curious about this, but need to learn more. What does this look like in other places?

**Comment by violasong** [comment]
Another thought I had - in the sidebar, Request URL, Method, and Status Code are a tad redundant with what would hopefully be showing in the table view. (Especially if we try something like hiding more of the other columns when the sidebar is open.)

**Comment by violasong** [comment]
I noticed Safari does this nice color coding. Does categorizing the headers this way make sense?


<img width="361" alt="image" src="https://user-images.githubusercontent.com/55994/78075200-f13f5f80-7358-11ea-8b22-a8c1eed5a91e.png">

**Comment by violasong** [comment]
Ah, I looked at the Safari screenshot more closely and realized it just has more redundant info that's separated out. I like the visual styling though.

**Comment by digitarald** [comment]
> Another thought I had - in the sidebar, Request URL, Method, and Status Code are a tad redundant with what would hopefully be showing in the table view. (Especially if we try something like hiding more of the other columns when the sidebar is open.)

I think I know what you mean. With the current table there is some redundance, but only "some" as the columns are often squished so small that the information can't really be seen. When we get around to minimize the table's sidebar-open mode to a single column this should not be an issue though, right?

> States for pending request, streamed request/response

**Pending requests** will only have _request_ parts (headers, payload, cookies, etc), but no response. Same for blocked requests btw. **Streamed responses** have some _response_ parts and others are still coming in in chunks. There are also streamed requests, for multipart uploads; which have chunk request data in multiple parts.

**Comment by digitarald** [comment]
> Ah, I looked at the Safari screenshot more closely and realized it just has more redundant info that's separated out. I like the visual styling though.

As usual, I don't understand Safari's color/icon choices ;) . But as long as things are readable and copy-able we are good. It feels similar to Paw's information hierarchy, which highlights key aspects in slightly bolder form and then just lists out the rest.

**Comment by violasong** [comment]
>With the current table there is some redundance, but only "some" as the columns are often squished so small that the information can't really be seen. When we get around to minimize the table's sidebar-open mode to a single column this should not be an issue though, right?

True, especially if only show the filename. But I wonder if we might want to keep the status/method in the table. 

>Pending requests will only have request parts (headers, payload, cookies, etc), but no response. Same for blocked requests btw. Streamed responses have some response parts and others are still coming in in chunks. There are also streamed requests, for multipart uploads; which have chunk request data in multiple parts.

I see - so, assuming we can differentiate whether some data is either missing or waiting, we could indicate that.

**Comment by digitarald** [comment]
> True, especially if only show the filename. But I wonder if we might want to keep the status/method in the table.

This reminds of of an idea I liked UX table, [Row to Details](https://uxdesign.cc/design-better-data-tables-4ecc99d23356#1347). It compresses additional details into the remaining row.

> I see - so, assuming we can differentiate whether some data is either missing or waiting, we could indicate that.

Yes, showing missing data as blank state (vs just not having the fields) makes sense.

Using phases from [Resource Timing](https://developer.mozilla.org/en-US/docs/Web/API/Resource_Timing_API/Using_the_Resource_Timing_API) to map out which data gets available over time:

1. `requestStart`: Request headers available (for many failed/blocked requests, this is we ever have)
2. `multipart/form-data` requests *(only important for file uploads)*: Chunked data are being added as they are sent
3. `responseStart`: Response headers available
4. `responseStart` to `responseEnd`: Response data chunks (and additional headers) available as they are received
5. `responsEnd`: All data available

@janodvarko did I miss a state?

**Comment by violasong** [comment]
> Content for the summarized section
> 
>     * Single-line summaries for
>       * Request: Method + Host + Path → Link to _Request_ tab
>       * Response: HTTP Code → Link to _Request_ tab
>       * HTTP/TLS Version → Link to _Security_
>       * Size / Transferred Size + Time to Load → Link to _Timing_
>       * Blocked Cookies? → Link to _Cookies_
>       * Referrer Policy

Some questions about this:

- What do you mean by "Link to Request tab"? (Different from the Request Headers section?)
- Should we also have a link to Cookies when it isn't a blocked request?
- Do we still want to show Address?

At first I was thinking of a menu-like UI like this

<img width="368" alt="image" src="https://user-images.githubusercontent.com/55994/78196830-7483b280-7438-11ea-80b7-be7b01a76c01.png">
 
But the sections that these would link to don't seem obviously connected to me. It might be better to name them, e.g.

<img width="361" alt="image" src="https://user-images.githubusercontent.com/55994/78197134-52d6fb00-7439-11ea-882e-ff77f3c6a04c.png">

**Comment by janodvarko** [comment]
>> I see - so, assuming we can differentiate whether some data is either missing or waiting, we could indicate that.

> Yes, showing missing data as blank state (vs just not having the fields) makes sense.

> @janodvarko did I miss a state?

I think that we can see it a bit more precisely as follows:
(but, the impact on the UI might be minimal)

* Request Start - Request headers available, POST data, request cookies, stack trace
* Sending POST Data - might take a time in case of big upload data
* Waiting for Response - might take a time if the server is slow. The server might be also in error state and no responsive (aka timeout)- in which case we might get no more data
* Response start - getting response headers, response cookies,
* Receiving data (aka downloading) Response data chunks (and additional headers) available as they are received
* Response end (or Request closed) - All data available now, no more data should be expected.

More info in HAR spec: http://janodvarko.cz/blog/har-12-spec/#timings
 

One more note. Request HTTP status is using a styled number in the first column
![image](https://user-images.githubusercontent.com/37785/78231062-874ac580-74d2-11ea-9cf1-b09230184ad5.png)

We can reuse the same style in the Headers panel.

Honza

**Comment by digitarald** [comment]
Related, I just checked out Safari's Network details pane which has some related ideas to grouping and labels (It is a bit hidden in Sources, which seems a nice touch, to have those information available in other panels; but kind of hard to discover):

<img width="305" alt="image" src="https://user-images.githubusercontent.com/8599/78302552-db17d780-74ef-11ea-8bbf-f85f9e2e464d.png">

**Comment by digitarald** [comment]
@violasong looks great.

@bomsy looking at Victoria's work, it doesn't seem like we'd want to use the tree component for that. Or do you see that we should customize it to present data like that?

> What do you mean by "Link to Request tab"? (Different from the Request Headers section?)

Sorry, the `Params` payload tab, which should be called `Request` for consistency.

> Should we also have a link to Cookies when it isn't a blocked request?

Not sure where I would put this, so no.

> Do we still want to show Address?

It would be important to know which server is resolved for some requests. See my analysis for Safari – maybe it fits with additional URL data.


> But the sections that these would link to don't seem obviously connected to me. It might be better to name them, e.g.

I like the option as well. Maybe they could appear on hover to reduce UI density?

Regarding the labels, in the [application panel design](https://www.figma.com/file/bOSqe1NWMrWWSS0em5LCfP/Application-panel?node-id=0%3A1), we had more muted labels. Maybe we can try those out for comparison.

**Comment by violasong** [comment]
> I like the option as well. Maybe they could appear on hover to reduce UI density?

That seems good, although maybe we want something there to encourage users to use the more contextual UI instead of looking through the tabs?

> Regarding the labels, in the application panel design, we had more muted labels. Maybe we can try those out for comparison.

That subtler look could work if we keep the labels short to allow for more spacing, like this:

<img width="368" alt="image" src="https://user-images.githubusercontent.com/55994/78399582-02cb7600-75aa-11ea-867d-0323e641e0cb.png">

It wouldn't match the request/response headers though.

([Figma](https://www.figma.com/file/rTeLZ1cmvHbgfJt91AloNi/DevTools-Network-Details-Pane-Copy?node-id=59%3A3) for all mockups so far)

>>> * Blocked Cookies? → Link to _Cookies_
>>  Should we also have a link to Cookies when it isn't a blocked request?
> Not sure where I would put this, so no.

What should show up in the summary for Blocked Cookies?

**Comment by violasong** [comment]
Ideas:

Expand arrow on the URL opens to show query string params

Some syntax coloring on the URL (this doesn't look great, but maybe something similar could work

<img width="926" alt="image" src="https://user-images.githubusercontent.com/55994/78732536-bbb9e980-78f7-11ea-9b7b-9f5859ca164e.png">

**Comment by violasong** [comment]
Idea:

More separation for warning/error messages in Header Summary

<img width="371" alt="image" src="https://user-images.githubusercontent.com/55994/78743887-dea6c680-7914-11ea-81df-a7f5d5a88e77.png">
Or should we just put it at the top?

<img width="364" alt="image" src="https://user-images.githubusercontent.com/55994/78743920-f54d1d80-7914-11ea-8039-6e951f0a7fb5.png">

**Comment by violasong** [comment]
Idea: In header sections, show info (MDN link) and copy button on hover

<img width="371" alt="image" src="https://user-images.githubusercontent.com/55994/78744060-64c30d00-7915-11ea-96fe-eeac4c341014.png">

**Comment by violasong** [comment]
Idea: All the above, with Response headers in blue sans-serif 12px labels + black monospace 11px values.

<img width="369" alt="image" src="https://user-images.githubusercontent.com/55994/78744800-5f66c200-7917-11ea-8e58-928a445646ab.png">

**Comment by violasong** [comment]
Same as above, except with gray labels. 

<img width="377" alt="image" src="https://user-images.githubusercontent.com/55994/78745036-ea47bc80-7917-11ea-9603-f610b7439453.png">

**Comment by janodvarko** [comment]
Thanks Victoria for the mockups, it looks really great!

* Expandable URL - to show URL params. We can also play with showing individual URL parts (domain, path, etc.)
* Color syntax for collapsed URL - awesome idea!
* Using block layout for the summary info is great (I don't think it's a problem that headers use different layout)
* The tracking protection notification message at the top is nice. We could use this concept for other messages as well (more info about request error state, etc)

Questions:
* What's supposed to be in the Resend dropdown? I can think of "Edit and Resend" and "Resend", is that it?
* What's suppoed to be in the Block dropdown?

Honza

**Comment by digitarald** [comment]
> Idea: All the above, with Response headers in blue sans-serif 12px labels + black monospace 11px values.

@violasong the smaller font seems also good to save space for longer output. Blue is also good since the summary section already has a grid layout to differentiate visually.

It just occurred to me that you mentioned another example for this, how the Debugger sizes those lines. Is that sizing/spacing aligned? It seems like it:

<img width="234" alt="image" src="https://user-images.githubusercontent.com/8599/78841237-b28d5300-79b1-11ea-95a1-7c39e274c4d0.png">


> The tracking protection notification message at the top is nice. We could use this concept for other messages as well (more info about request error state, etc)

What other errors did you have in mind, @janodvarko?

An alternative to the error could be adding a line with a warning in-line with the tracking category.

```
Referrer:         no-referrer-its-referer
Content Blocking: Tracking Content
                  🛡 Be aware that cookies or content for this URL could be blocked. (?)`
```

WDYT, @violasong ?

> Color syntax for collapsed URL - awesome idea!

@violasong I like the idea. It does make it a bit more readable. I am not sure which parts we should highlight; as use cases vary on what's important to users. Have you seen an example of that anywhere else?

> What's supposed to be in the Resend dropdown? I can think of "Edit and Resend" and "Resend", is that it?

Yes, that was my proposal, making sure we are exposing more options from the context meny in the toolbar. @violasong another idea would be an "Edit" button and a "Resend" button; where Edit implies that it would be send again anyways. So we avoid the dropdown.

> What's suppoed to be in the Block dropdown?

"Block URL" is very blunt atm. I proposed this as we can expose more options. "Block Domain" and "Block Path" could be added here. To simplify, it is probably easier to use the blocking icon from the network toolbar and avoid the dropdown. If users want to fine-tune, the blocking sidepanel will open automatically anyways and they can edit the URL down to what they need.

**Comment by janodvarko** [comment]
> What other errors did you have in mind, @janodvarko?

I like the idea to use the notification box as a generic way to show any error/warning/info that's worth of user attention.

* Blocked requests - blocked reason, Blocked Extension (this info is hard to read from a column).
* Classification flag - cryptoming/tracking/fingerpriting
* Tracking protection - Detailed explanation

Honza

**Comment by violasong** [comment]
Updated color syntax for collapsed URL:

<img width="919" alt="image" src="https://user-images.githubusercontent.com/55994/78951887-809af000-7a88-11ea-8c4a-992bfd404008.png">

Changed magenta to gray which I think works better, and added in the filter bar.

> @violasong I like the idea. It does make it a bit more readable. I am not sure which parts we should highlight; as use cases vary on what's important to users. Have you seen an example of that anywhere else?

I don't think I've seen this elsewhere, though the grayed out https:// is like Firefox's URL bar. The idea I had is that the blue parts correlate to the blue keys when expanded.

**Comment by violasong** [comment]
>Expandable URL - to show URL params. We can also play with showing individual URL parts (domain, path, etc.)

So something like this?

<img width="481" alt="image" src="https://user-images.githubusercontent.com/55994/78952742-6b739080-7a8b-11ea-8848-463da8bdb784.png">

**Comment by violasong** [comment]
Mockup with:
- Harald's last suggestion for error message. This de-emphasizes the error and makes it more a part of the summary rather than an extra section. Sounds like that's what we want?
- A more typically-sized URL with no params, shown in the style of the query param examples
- Smaller toggle for switching to raw

<img width="387" alt="image" src="https://user-images.githubusercontent.com/55994/79028048-7c7fd880-7b43-11ea-98f9-b6eb3388039b.png">

**Comment by digitarald** [comment]
@violasong looks really nice! Makes me want to click around and play with it.

One thing I am not sure about is the shortened labels ("Blocking" is less descriptive than "Content Blocking") and other languages will vary in length as well. What's the strategy to make this responsive and overflow in sensible ways?

> Harald's last suggestion for error message. This de-emphasizes the error and makes it more a part of the summary rather than an extra section. Sounds like that's what we want?

> A more typically-sized URL with no params, shown in the style of the query param examples

Should this URL be limited in lines, using `line-clamp`?

FYI, the parts of the URL that we should show are all listed in https://developer.mozilla.org/en-US/docs/Web/API/URL .

> Smaller toggle for switching to raw

Perfect size!

Just to get consensus (I am a fan of hover menus), is the idea from https://github.com/firefox-devtools/ux/issues/116#issuecomment-610742996 accepted? @bomsy @janodvarko

**Comment by violasong** [comment]
> One thing I am not sure about is the shortened labels ("Blocking" is less descriptive than "Content Blocking") and other languages will vary in length as well. What's the strategy to make this responsive and overflow in sensible ways?

Ah, I see. I'll look at this some more. Here's one thing we can do with the alignment, though we don't do this anywhere else.

<img width="459" alt="image" src="https://user-images.githubusercontent.com/55994/79281918-a8170180-7e68-11ea-8eb1-7978d60d2c00.png">


>Should this URL be limited in lines, using line-clamp? 

I thought we were saying that the full URL is important here, as this is the only place to see the whole thing (besides the query string, but that's an extra step and kind of different).

**Comment by digitarald** [comment]
> I thought we were saying that the full URL is important here, as this is the only place to see the whole thing (besides the query string, but that's an extra step and kind of different).

Right, makes sense. Might also work if we expand it to a full URL when the user expands this section. If we do a proper line-break (not like now), 1000+ character URLs will take up a lot of space.

**Comment by bomsy** [comment]
> Just to get consensus (I am a fan of hover menus), is the idea from #116 (comment) accepted? 
@digitarald I like that idea, out of curiosity how would it look for headers that have the warning icons (set-Cookie in this example)?

**Comment by violasong** [comment]
>  Might also work if we expand it to a full URL when the user expands this section. If we do a proper line-break (not like now), 1000+ character URLs will take up a lot of space.

I see what you mean, there needs to be some level of truncation at some point :D 

This could work, though the expanded query string part could end up being pretty massive

<img width="920" alt="image" src="https://user-images.githubusercontent.com/55994/79375333-4d83b100-7f0d-11ea-802b-bb7c489de17e.png">

**Comment by digitarald** [comment]
> @digitarald I like that idea, out of curiosity how would it look for headers that have the warning icons (set-Cookie in this example)?

Maybe @violasong has ideas how some icons are always on and don't need hover. Otherwise we could show errors like we do them inline in the general section?

**Comment by digitarald** [comment]
> This could work, though the expanded query string part could end up being pretty massive

As it's expanded optionally, that seems acceptable.

**Comment by violasong** [comment]
> I like that idea, out of curiosity how would it look for headers that have the warning icons (set-Cookie in this example)?

Good point. I realized I also didn't work out how the hovering would work on a long row. 

One quick idea: buttons float on top on hover, like in Copy Rules:

<img width="749" alt="image" src="https://user-images.githubusercontent.com/55994/79400743-e92c1600-7f3b-11ea-9c4f-f4bc83a9ba24.png">

**Comment by violasong** [comment]
Hm, right after posting this I realized it's probably too much fanciness as it obscures basic copy paste, like if someone just wanted to grab the times part of that timestamp.

So the other option is there's just an area on the right side saved for the icons, and the text would wrap before getting there.

**Comment by digitarald** [comment]
> One quick idea: buttons float on top on hover, like in Copy Rules:

I like that. One caveat to test during implementation is how hover-enabled buttons interact with selecting text portions.

**Comment by fvsch** [comment]
For the URL params when there are many:

> This could work, though the expanded query string part could end up being pretty massive

One option is to truncate and, rather than expand, add a button that sends the user to the "Params" tab where we already parse and display the URL parameters.

**Comment by digitarald** [comment]
@violasong One aspect to think about is long header names and how to align the values:

<img width="306" alt="image" src="https://user-images.githubusercontent.com/8599/82386556-94bc0080-99e9-11ea-9711-c959637cb4b3.png">

**Comment by digitarald** [comment]
Just for reference, leaving this example for a super-long URLs that makes the URL preview take over the whole above the fold space for the panel; so we can test it against the new bug:

<img width="327" alt="image" src="https://user-images.githubusercontent.com/8599/82387001-b4075d80-99ea-11ea-97d7-c4feec8fb968.png">

**Comment by violasong** [comment]
> One aspect to think about is long header names and how to align the values:

Ah, this screenshot is unfortunate; makes me think we should full-length wrap the values.

For super long URLs, an idea I just talked about with @nchevobbe regarding the Send Request Figma was limiting it to a certain number of rows (5?), and adding scroll overflow once it goes over.

**Comment by digitarald** [comment]
> adding scroll overflow once it goes over.

[line-clamp](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp) to the rescue.
